src/components/TaskListScreen.js

```diff
@@ -0,0 +1,134 @@
+            description: newTaskDescription,
+            customer: profile.reference,
+            created_at: firebase.firestore.Timestamp.fromDate(new Date())
+        }).then(function () {
+            console.log("Task created!");
+            history.push("/tasks");
+        }).catch(function (error) {
+            console.error("Error writing document: ", error);
+        });
+    }
+
+    return (
+        <Switch>
+            <Route exact path="/tasks">
+                {profile && profile.customer ?
+                    <Link to="/tasks/add" className="add-task">
+                        <img src={addIcon} alt="list" />
+                    </Link> : null}
+                <div className="overflow-auto">
+                    {profile ? tasks.map((task, index) => (
+                        <Link className="card task-card" key={index} to={"/tasks/" + task.id}>
+                            <div className="flex-row">
+                                <TextAvatar width="40" height="40" text={task.customer} />
+                                <div className="flex-column flex-1 justify-between">
+                                    <span className="semi-bold">{task.name}</span>
+                                    <span className="regular">{task.customer}</span>
+                                </div>
+                            </div>
+                            <div className="flex-row justify-between bottom">
+                                <span className="regular">{task.repliesCount} заявок</span>
+                                <span className="regular">{moment(task.created_at).fromNow()}</span>
+                            </div>
+                        </Link>
+                    )) : null}
+                </div>
+            </Route>
+            <Route path="/tasks/add">
+                {profile && profile.customer ?
+                    <div className="card flex-1 flex-column text-start add-task-wrapper">
+                        <span className="semi-bold background">Создание нового заказа</span>
+                        <input className="form-input" type="text" placeholder="Придумайте название заказа..."
+                            value={newTaskName} onChange={(event) => {
+                                setNewTaskName(event.target.value)
+                            }} />
+                        <textarea className="flex-1 form-input" placeholder="Опишите Ваш заказ..."
+                            value={newTaskDescription} onChange={(event) => {
+                                setNewTaskDescription(event.target.value)
+                            }} />
+                        <button className="button" onClick={CreateTask}>Создать заказ</button>
+                    </div>
+                    : <Redirect to="/tasks" />}
+            </Route>
+            {profile ? <Route path="/tasks/:id" children={<TaskScreen />}></Route> : <Redirect to="/tasks" />}
+        </Switch>
+    );
+}
+
```

Comment:
There is a lot of code, which makes `component` big and messy. Every big-sized function should be moved into `hooks` which produce this function, and inside this `hook-function` should be used `helpers` or `utils`.
Also, best practices say, what every component, which connected with more than 1 `UI` layer (e.g. `UI + Network`), should be split into `component or view` (component, which returns `UI`) and `container`, which connected with `Network` layer or some business logic, and returns `component or view`, and fills component with `props`